engine-tests/src/tests/xcc.rs

```diff
@@ -589,6 +605,17 @@ fn test_xcc_schedule_gas() {
 fn test_xcc_exec_gas() {
     let mut router = deploy_router();
 
+    let create_promise_chain =
```

Comment:
Just a comment, good practice to have named defined functions which are self-descriptive. Good work!